mne/chpi.py

```diff
@@ -366,9 +378,180 @@ def _time_prefix(fit_time):
 
 
 @verbose
+def _fit_cHPI_amplitudes(raw, time_sl, hpi, fit_time, verbose=None):
+    """Fit amplitudes for each channel from each of the N cHPI sinusoids.
+
+    Returns
+    -------
+    sin_fit : ndarray, shape (n_freqs, n_channels)) or None :
+        The sin amplitudes matching each cHPI frequency
+            or None if this time window should be skipped
+    """
+    with use_log_level(False):
+        # loads good channels
+        meg_chpi_data = raw[hpi['meg_picks'], time_sl][0]
+
+    mchpi_data = np.mean(meg_chpi_data, axis=1, keepdims=True)
+    this_data = meg_chpi_data - mchpi_data
+
+    # which HPI coils to use
+    # other then erroring I don't see this getting used elsewhere?
+    if hpi['hpi_pick'] is not None:
+        with use_log_level(False):
+            # loads hpi_stim channel
+            chpi_data = raw[hpi['hpi_pick'], time_sl][0]
+
+        ons = (np.round(chpi_data).astype(np.int) &
+               hpi['on'][:, np.newaxis]).astype(bool)
+        n_on = np.sum(ons, axis=0)
+        if not (n_on >= 3).all():
+            logger.info(_time_prefix(fit_time) + '%s < 3 HPI coils turned on, '
+                        'skipping fit' % (n_on.min(),))
+            return None
+        # #TODO REMOVE # ons = ons.all(axis=1)  # which HPI coils to use
```

Comment:
This is commented out in master. It seems like the idea was to only fit HPIs that were on for a particular window but it isn't clear how that plays out in the rest of the code. Do you have a test that has 4 HPIs for some windows and only 3 for others?